src/vs/workbench/contrib/notebook/browser/diff/notebookDiffEditor.ts

```diff
@@ -601,11 +604,39 @@ export class NotebookTextDiffEditor extends EditorPane implements INotebookTextD
 		return isSame;
 	}
 
-	private _setViewModel(viewModels: DiffElementViewModelBase[]) {
+	private readonly viewModelStore = this._register(new DisposableStore());
+	private _setViewModel(viewModels: DiffElementCellViewModelBase[]) {
 		this._diffElementViewModels = viewModels;
-		this._list.splice(0, this._list.length, this._diffElementViewModels);
+		this.viewModelStore.clear();
+
+		const newViewModel: IDiffElementViewModelBase[] = [];
+		let placeholder: DiffElementPlaceholderViewModel | undefined = undefined;
```

Comment:
This is hacky, will need to create a class around this to make this simpler,
but this was the POC, rest seems about the state i want it in, still need to polish and fix some parts